### PR TITLE
refactor!: remove the `https` and the `http2` option

### PR DESCRIPTION
**Pull request by snitin315**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [x] This is a **code refactor**
- [x] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

Removed tests.

### Motivation / Use-Case

Remove the deprecated  `https` and `http2` option

### Breaking Changes

BREAKING CHANGES:  the deprecated  `https` and `http2` options were removed in favor of the `server` option

### Additional Info
No

### CODE REVIEW

**Comment by snitin315** [comment]
I'll fix CI.

File: test/e2e/http2.test.js [hunk 1]
```diff
@@ -1,238 +0,0 @@
...
-      expect(await response.text()).toMatchSnapshot("response text");
-
-      expect(consoleMessages.map((message) => message.text())).toMatchSnapshot(
-        "console messages"
-      );
-
-      expect(pageErrors).toMatchSnapshot("page errors");
-    });
-  });
-});
```
Comment (alexander-akait) [inline test/e2e/http2.test.js:238]: Just for check - do we have the same test cases for the other options (same for `https`)

Comment (snitin315) [reply test/e2e/http2.test.js:238]: Yes, all tests cases are present for the `server` option.

**Review by alexander-akait — APPROVED** [review]

**Comment by alexander-akait** [comment]
Thank you, I think we will write the small migration guide before release to avoid many question and issues

**Pull request merged**